onnx/defs/generator/defs.cc

```diff
@@ -523,4 +527,185 @@ ONNX_OPERATOR_SET_SCHEMA(
           updateOutputShape(ctx, 0, {batch_size, sample_size});
         }));
 
+static const char* Range_ver11_doc = R"DOC(
+Generate a tensor containing a sequence of numbers that begin at `start` and extends by increments of `delta` 
+up to `limit` (exclusive).)DOC";
+
+template <typename T>
+inline int compute_output_dim_for_range(
+    const TensorProto* start,
+    const TensorProto* limit,
+    const TensorProto* delta) {
+  const auto& start_data = ParseData<T>(start);
+  const auto& limit_data = ParseData<T>(limit);
+  const auto& delta_data =
+      delta != nullptr ? ParseData<T>(delta) : std::vector<T>(1, 1);
+
+  if (start_data.size() != 1 || limit_data.size() != 1 ||
+      delta_data.size() != 1) {
+    fail_shape_inference(
+        "Input to 'Range' op should be scalars (Tensor with only one element)");
+  }
+
+  int n = static_cast<int>(
+      ceil((1.0 * (limit_data[0] - start_data[0])) / delta_data[0]));
+  if (n < 0)
+    n = 0;
+
+  return n;
+}
+
+const std::vector<NodeProto> build_nodes_range_op() {
+    // body for 'Loop node'
+    GraphProto loop_sub_graph;
+    loop_sub_graph.set_name("loop body attribute");
+
+    // 'Loop' node 'body' attribute's graph inputs
+    auto* input_value_info_proto_0 = loop_sub_graph.add_input();
+    input_value_info_proto_0->set_name("i");
+
+    auto* input_value_info_proto_1 = loop_sub_graph.add_input();
+    input_value_info_proto_1->set_name("cond");
+
+    auto* input_value_info_proto_2 = loop_sub_graph.add_input();
+    input_value_info_proto_2->set_name("prev");
+
+    // 'Loop' node 'body' attribute's graph nodes
+    auto* node_proto_0 = loop_sub_graph.add_node();
+    node_proto_0->set_op_type("Identity");
+    node_proto_0->add_input();
+    node_proto_0->set_input(0, "cond");
+    node_proto_0->add_output();
+    node_proto_0->set_output(0, "cond_out");
+
+    auto* node_proto_1 = loop_sub_graph.add_node();
+    node_proto_1->set_op_type("Add");
+    node_proto_1->add_input();
+    node_proto_1->set_input(0, "prev");
+    node_proto_1->add_input();    
+    node_proto_1->set_input(1, "delta");
+    node_proto_1->add_output();
+    node_proto_1->set_output(0, "current");
```

Comment:
Is it `current` or `prev`?

Reply:
If my understanding of the `Loop` operator is correct,I believe this has to be `current`. `current` is then mapped to the loop carried dependency `prev` below.

Reply:
You're probably right. My question comes from an statement in Loop's spec.
```
Any variables which you wish to make available in the enclosing scope (i.e. the variables b and keepgoing) must be declared as either loop-carried dependencies (_both_ at the op inputs and output and at the body net input and output) or scan_outputs.
```
which makes me wonder if we really replace `current` with `prev` in your code below
```cpp
    auto* output_value_info_proto_1 = loop_sub_graph.add_output(); 
    output_value_info_proto_1->set_name("current");
```
.

Reply:
This is a goo point. Honestly, I do not know if we ca make "current" -> "prev" in the specific line that you mention.  From the spec, I understood that the subgraph's outputs are: condition, loop carried dependencies (final values), and scan outputs. Since "cond" is made "cond_out" at the output, I guess it doesn't hurt to make "prev" into "current" at the output.  @gramalingam  - thoughts please?

P.S. - I didn't have a lot to go by when I wrote this implementation. I did this from what I could understand from the spec. Please suggest any simplifications if you can in the loop body...

Reply:
Yes. Let's me add more details to make sure we are on the same page. The loop-carried dependencies are matched and assigned by position in the body graph's input and output lists. If we have iteration count `T`, loop-terminating condition `cond`, and loop-carried dependencies `[X, Y, Z]`, the input lists of the body graph would be `[T, cond, X, Y, Z]`, and its output list would be `[cond_new, X_new, Y_new, Z_new]`. The variable `cond_new`/`X_new`/`Y_new`/`Z_new` would be assigned to `cond`/`X`/`Y`/`Z` at the beginning of the next iteration.

Reply:
That is what I get from the spec too. So, "current" will be fed in as "prev" in the second iteration onwards. For the initial iteration, "start" will be fed in as "prev". Based, on "cond" in the first iteration, "start" will be included in the result (cond is true) or may not be (cond is false)